packages/components/src/ObjectViewer/JSONLike/JSONLike.component.js

```diff
@@ -428,10 +451,32 @@ export function Item(props) {
 						onEdit={props.onEdit}
 						onChange={props.onChange}
 						className={props.nativeValueClassName}
+						wrap={nativeValueWrap}
 					/>
 				}
 				type={props.showType ? info.type : null}
 				tag={tag}
+				setWidth={lineItemWidth}
+				// icon is shown when LineItem value is a long field and needs to be wrapped
+				icon={
+					lineItemWidth && (
+						<Action
+							key="toggle"
+							className={classNames(theme.chevron, { [theme['chevron-filled']]: rotation })}
+							icon={'talend-chevron-left'}
+							iconTransform={rotation ? 'rotate-90' : 'rotate-270'}
+							onClick={e => {
+								e.stopPropagation();
+								setNativeValueWrap(val => !val);
+								setRotation(val => !val);
```

Comment:
`rotation` and `nativeValueWrap` are the same. You can just use `valueWrap`, if it's wrapped, then you rotate the icon